### PR TITLE
Consolidate logic for Pin image

### PR DESCRIPTION
**Pull request by ntascii**
## Summary 
Always show Pin image. That’s the reflection of DB status.
- If there is golden - allow unpin (auto update) and pin
- If there is no golden  - disabled pin image
  - if cluster is pinned, no problem
  - if cluster is unpinned, automatically pin image
## Test plan 
Test locally and dev1
No golden, disable Pin Image checkbox and auto checked.
Screenshot 2023-03-10 at 12 03 37 PM
There is a golden, reflect db status 
 Auto update
Screenshot 2023-03-10 at 12 10 55 AM
Pin Image 
Screenshot 2023-03-10 at 12 09 38 PM
Save and it works.

### CODE REVIEW

**Review by tylerwowen — CHANGES_REQUESTED** [review]
Can you please remove formatting changes? There are too many unrelated change and can make code blame difficult.

File: deploy-board/deploy_board/templates/configs/new_capacity_adv.html [hunk 1]
```diff
@@ -88,300 +106,297 @@ <h4 class="panel-title pull-left">Capacity</h4>
...
+                    return {
+                        name: key,
+                        value: info.defaultCMPConfigs[key]
+                    }
+                }).filter(function (item) { return item.name != 'cmp_group' }),
+            assignPublicIP: false,
+            baseImageHelpData: [],
+            baseImages: mapBaseImagesToOptions(baseImagesSorted),
+            baseImageValue: defaultImageId,
+            pinImage: getPinImageValue(goldenImage, true), // Default auto update if there is a golden image
```
Comment (tylerwowen) [inline deploy-board/deploy_board/templates/configs/new_capacity_adv.html:146]: We should be careful not to publish Golden image util it's GA.

**Comment by ntascii** [comment]
I added the auto formating in the later commit so you can see the first commit for the change.

File: deploy-board/deploy_board/templates/configs/new_capacity_adv.html (hunk 1 above)
Comment (ntascii) [reply deploy-board/deploy_board/templates/configs/new_capacity_adv.html:146]: Good point. Default to Pin Image until GA will be better.

**Comment by ntascii** [comment]
Updated the default for auto update and reverted the auto format commit for reviewing easier.

**Review by tylerwowen — DISMISSED** [review]

**Review by tylerwowen — APPROVED** [review]

**Pull request merged**